src/client/preview/client_api.js

```diff
@@ -67,10 +99,10 @@ export default class ClientApi {
             return decorated(context);
           }, context);
         };
-      }, getStory);
+      }, storyFn);
 
       // Add the fully decorated getStory function.
-      this._storyStore.addStory(kind, storyName, fn);
+      this._storyStore.addStory(kindConfig, storyName, { storyFn: fn, storyMeta });
```

Comment:
It's a cleaner interface, but I'm unsure how this would affect other addons (react-storybook-addon-info etc) which also utilities more parameters. I'll take a look at it, and how it can be solved though. :)